objc/objc_runtime_darwin_test.go

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+package objc
+
+import (
+	"fmt"
+	"math"
+	"runtime"
+	"unsafe"
+
+	"github.com/ebitengine/purego"
+)
+
+func init() {
+	// this is here so that the program doesn't crash with:
+	// 		fatal error: schedule: in cgo
+	// this can be removed after purego#12 is solved.
+	// this is due to moving goroutines to different threads while in C code
+	runtime.LockOSThread()
+}
+
+func ExampleAllocateClassPair() {
+	var class = AllocateClassPair(GetClass("NSObject\x00"), "FooObject\x00", 0)
+	class.AddMethod(RegisterName("run\x00"), IMP(func(self ID, _cmd SEL) {
+		fmt.Println("Hello World!")
+	}), "v@:\x00")
+	class.Register()
+
+	var FooObject = ID(class).Send(RegisterName("new\x00"))
+	FooObject.Send(RegisterName("run\x00"))
```

Comment:
Just a qeustion: is the function `run` a instance method or a static method? If this is an instance method, `fooObject` as a Go variable is better than `FooObject`.

Reply:
These are instance methods